persistence/persistence-spark/src/main/java/org/cbioportal/persistence/spark/DiscreteCopyNumberSparkRepository.java

```diff
@@ -0,0 +1,207 @@
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta getMetaDiscreteCopyNumbersInMolecularProfileBySampleListId(String molecularProfileId, String sampleListId, List<Integer> entrezGeneIds, List<Integer> alterationTypes) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<DiscreteCopyNumberData> fetchDiscreteCopyNumbersInMolecularProfile(String molecularProfileId, List<String> sampleIds, List<Integer> entrezGeneIds, List<Integer> alterationTypes, String projection) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<DiscreteCopyNumberData> getDiscreteCopyNumbersInMultipleMolecularProfiles(List<String> molecularProfileIds, List<String> sampleIds, List<Integer> entrezGeneIds, List<Integer> alterationTypes, String projection) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta fetchMetaDiscreteCopyNumbersInMolecularProfile(String molecularProfileId, List<String> sampleIds, List<Integer> entrezGeneIds, List<Integer> alterationTypes) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<CopyNumberCountByGene> getSampleCountByGeneAndAlterationAndSampleIds(String molecularProfileId, List<String> sampleIds, List<Integer> entrezGeneIds, List<Integer> alterations) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<CopyNumberCountByGene> getSampleCountInMultipleMolecularProfiles(List<String> molecularProfileIds, List<String> sampleIds, List<Integer> entrezGeneIds, List<Integer> alterations) {
+        
+        // filter based on moleclarProfileIds
+        Dataset<Row> casels = parquetLoader.loadCaseListFiles(spark, new HashSet<>(molecularProfileIds), false);
+        casels = casels.withColumnRenamed("case_list_ids", "SAMPLE_ID");
+        if (!CollectionUtils.isEmpty(sampleIds)) {
+            casels = casels
+                .where(casels.col("SAMPLE_ID").isin(sampleIds.toArray()));
+        }
+        // get studyIds and filter CNA data based on alterations
+        List<String> studyIds = casels.select("cancer_study_identifier").distinct()
+            .collectAsList().stream().map(r -> r.getString(0)).collect(Collectors.toList());
+        Dataset<Row> cnaData = parquetLoader.loadStudyFiles(spark, new HashSet<>(studyIds), "/" + ParquetConstants.DATA_CNA, false);
+        if (!CollectionUtils.isEmpty(alterations)) {
+            cnaData = cnaData.where(cnaData.col("VALUE").isin(alterations.toArray()));
+        }
+        cnaData = cnaData.join(casels, "SAMPLE_ID");
+
+        String mainCol = Arrays.asList(cnaData.columns()).contains("Hugo_Symbol") ? "Hugo_Symbol" : "Entrez_Gene_Id";
```

Comment:
Each row id the data file may or may not have values for `Hugo_Symbol` and `Entrez_Gene_Id`, but at least one would be present. I think both should be handled.

Reply:
or missing `Hugo_Symbol` and `Entrez_Gene_Id` should be filled while creating those parquet files.

Reply:
Correct, I did test for when they have either hugoGeneSymbol or entrezGeneId. I was keeping the ParquetWriter simple for now, but as per your suggestion, made some changes so that ParquetWriter can connect to mysql database and get geneDetails (entrezGeneId, hugoGeneSymbol, and cytoband).